### PR TITLE
perf(runtime): speed up iterable_array_from for arrays

### PR DESCRIPTION
**Pull request by jonkwheeler**
## Summary

`iterable_array_from` accepted arrays but always walked `Symbol.iterator`, allocating an iterator and a `{ value, done }` result per element. Arrays now use `iterable.slice(index)`, which keeps holes in holey arrays.

Sets, raw iterators, and array-likes are unchanged.

## Audit notes

`@tsrx/runtime` is compiler helpers (refs, spreads, `@for` mapping), not a signals/effects runtime. Other hot paths were measured before landing this:

- `map_iterable` / `map_array` — already specialized for arrays. Preallocation, native `map`, and flatten variants were mixed or regressing across list sizes (10 / 100 / 1000) and vnode vs text vs fragment workloads.
- `normalize_spread_props` — rest-arg removal and indexed `ownKeys` were inside noise (~1–4%). This path was just tuned in #42.
- `normalize_spread_props_for_ref_attr` in-place `defineProperty` — ~0–3%, overlapping quartiles.
- `exclude_prop_from_object` `defineProperties` batch — **−20%**.

Only the array fast path in `iterable_array_from` was a clear, reproducible win.

## Benchmarks

Node `v22.14.0`, linux x64, isolated `perf_hooks` microbenchmarks (warmup 200ms, 400ms × 13 runs, median).

`Array.prototype.slice` vs the previous iterator walk (same-process bake-off):

| Workload | Before (iterator) | `slice` | Delta |
| --- | ---: | ---: | ---: |
| array 100, index 0 | 2,232,508 ops/s | 14,768,076 ops/s | **+561%** |
| array 100, index 10 | 2,266,330 ops/s | 15,465,162 ops/s | **+582%** |
| array 1000, index 0 | 233,020 ops/s | 3,150,961 ops/s | **+1252%** |

Set / iterator workloads stay on the existing path.

## Test plan

- [x] Runtime coverage for dense `slice`, index edge cases, holey-array holes, Set, iterator, and array-like
- [x] `pnpm exec vitest run` on language-helpers runtime and type tests (2 files, 15 tests) after the review follow-up

<div><a href="https://cursor.com/agents/bc-ee7b7d85-7d18-4e0e-a808-98b30561a2d4?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-ee7b7d85-7d18-4e0e-a808-98b30561a2d4&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: packages/tsrx-runtime/src/language-helpers.js [hunk 1]
```diff
@@ -57,14 +57,44 @@ export function array_slice(array_like, ...args) {
...
+ *
  * @template T
  * @param {Iterable<T> | Iterator<T> | ArrayLike<T>} iterable
  * @param {number} [index]
  * @returns {T[]}
  */
 export function iterable_array_from(iterable, index = 0) {
+	if (is_array(iterable)) {
+		return array_from_index(iterable, index);
+	}
```
Comment (leonidaz) [inline packages/tsrx-runtime/src/language-helpers.js:71]: we don't need an extra function, this should just be:

```suggestion
	if (is_array(iterable)) {
		return array.slice(index);
	}
```

**Review by leonidaz — CHANGES_REQUESTED** [review]

File: packages/tsrx-runtime/src/language-helpers.js (hunk 1 above)
Comment (leonidaz) [reply packages/tsrx-runtime/src/language-helpers.js:71]: the reason is if it's even a holey array, we still want holes to be present, we're not changing that.

**Review by leonidaz — APPROVED** [review]
